platform.py

```
@@ -14,239 +14,502 @@
-                    install_tool(TOOL, retry_count + 1)
+            if result.returncode != 0:
+                logger.error("idf_tools.py installation failed")
+                return False
+
+            logger.debug("idf_tools.py executed successfully")
+            return True
+
+        except subprocess.TimeoutExpired:
+            logger.error(f"Timeout in idf_tools.py after {SUBPROCESS_TIMEOUT}s")
+            return False
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)